src/gameList/GameListScreen.js

```diff
@@ -18,25 +18,41 @@ export default class GameListScreen extends React.Component {
         loading: true,
     };
 
-    static navigationOptions = {
-        title: 'Join a game',
+    static navigationOptions = ({ navigation }) => {
+        const listType = navigation.state.params.listType
+            ? navigation.state.params.listType
+            : 'Play';
+        let title = '';
+        switch (listType) {
+            case 'current':
+                title += 'Continue';
+                break;
+            case 'opened':
+                title += 'Join';
+                break;
+            case 'onlywatch':
+                title += 'Watch';
+        }
+        title += ' a game';
+        return { title };
     };
 
     static propTypes = {
         navigation: PropTypes.object.isRequired,
     };
 
     async componentDidMount() {
+        const { listType } = this.props.navigation.state.params;
         this.setState({ loading: true });
-        const games = await listGames();
+        this.setState({ listType: listType });
```

Comment:
I would initialize the state at L19 with the value from the `navigation` prop. Not very important though